### PR TITLE
Update ref category page for serial

### LINKED ISSUE #2101: Serial documentation blocks under advanced section don't render
**Describe the bug**
Although the blocks exist in the markdown. Nothing can be found under the Advanced section of the page. May be related to issue #2098. 

**To Reproduce**
1. Go to https://makecode.microbit.org/reference/serial

**Expected behavior**

**Screenshots**

Serial docs
image

Serial markdown
image

**Desktop (please complete the following information):**
 - OS: Chrome OS
 - Browser: Chrome

### PR DESCRIPTION
**Pull request by ganicke**
Remove the TX and RX buffer size APIs and update the sig usage for serial ref docs.

Fixes #2101.

### CODE REVIEW

File: docs/reference/serial.md [hunk 1]
```diff
@@ -31,7 +29,5 @@ serial.setTxBufferSize(64);
 [writeString](/reference/serial/write-string), 
 [writeNumbers](/reference/serial/write-numbers), [readUntil](/reference/serial/read-until), [readLine](/reference/serial/read-line),
 [readString](/reference/serial/read-string), [onDataReceived](/reference/serial/on-data-received),
-[redirect](/reference/serial/redirect), [writeBuffer](/reference/serial/write-buffer), [readBuffer](/reference/serial/read-buffer),
-[redirectToUSB](/reference/serial/redirect-to-usb),
-[set rx buffer size](/reference/serial/set-rx-buffer-size),
```
Comment (pelikhan) [inline docs/reference/serial.md]: why are we removing those references? this API exist in typescript.

**Comment by abchatra** [comment]
These are JavaScript only API?

**Comment by ganicke** [comment]
Define the blocks again and put them under `... more`?

image

**Comment by pelikhan** [comment]
Yes advanced section. This is a commonly requested feature because the serial buffer is very small by default.

________________________________

**Comment by ganicke** [comment]
Closing in favor of #2128.

**Pull request closed without merging**